src/Internal/DirectoryCreator.php

```diff
@@ -0,0 +1,37 @@
+<?php declare(strict_types = 1);
+
+namespace PHPStan\Internal;
+
+use InvalidArgumentException;
+use function clearstatcache;
+use function is_dir;
+use function mkdir;
+
+final class DirectoryCreator
+{
+
+	/**
+	 * @throws InvalidArgumentException
+	 */
+	public static function ensureDirectoryExists(
+		string $directory,
+		int $mode = 0755,
+		bool $recursive = true,
+	): void
+	{
+		if (is_dir($directory)) {
```

Comment:
since we have to use error suppresing in the below `mkdir()` calls, IMO we should not use `is_dir` here, which produces unnecessary IO

Reply:
That defeats the whole purpose of me submitting this PR in the first place... yes error suppression is used but an unnecessary warning is still generated which can cause issues for clients. There have been previous issues and PRs to add the `is_dir` check to avoid the warning being generated

Reply:
I've previously told you the root issue needs to be fixed in Pest. PHPStan's source code needs to make sense on its own. Otherwise it's not really good engineering and the code would be very brittle.

Reply:
I can remove, but I'm just pointing out that I'm not the first person this has caused issues with and who has tried to add the dir existence check

Reply:
removed